test/fixture.cpp

```diff
@@ -88,6 +92,29 @@ TestFixture::TestFixture(const char * const _name)
 
 bool TestFixture::prepareTest(const char testname[])
 {
+    const char* coverage = std::getenv("COVERAGE");
+    if (coverage) {
+        std::string testDir = "./coverage_per_test/" + std::string(testname); // Adjust the path as needed
+        if (mkdir(testDir.c_str(), 0777) != 0) {
```

Comment:
This is a big no-no. You should not create files or directory with execute or write permissions if not necessary. That might lead to security vulnerabilities. It should be `0644` at most.